public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -57,6 +57,7 @@ interface Props<TQuery extends DataQuery> {
   history?: Array<HistoryItem<TQuery>>;
   eventBus?: EventBusExtended;
   alerting?: boolean;
+  onTriggerTracking?: (action: string, queryStatus?: boolean | undefined) => void;
```

Comment:
I think the method shouldn't be called `onTriggerTracking` as it can be used to handle anything (maybe something like `onEvent` would be better?). Also the optional `queryStatus` param is confusing because it's passed only with `enableDisable` so it's harder to understand just reading the API. @Elfo404, @ivanortegaalba please correct me if I'm wrong by I think what you  meant was to have separate handlers for each event:

```typescript
onQueryRemoved: () => void;
onQueryCopied: () => void;
onQueryToggled: (shown: boolean): void;
```

Reply:
Done.